csrc/cpu_ops.cpp

```diff
@@ -1,5 +1,7 @@
 #include <BinSearch.h>
-#include <pthread.h>
+#include <thread>
+#include <vector>
```

Comment:
NOT TESTED! Refactored to use the new std::async job system. Hopefully cross platform and more efficient than starting 256 threads :)